### PR TITLE
Use six.* to trim down compat.py

### PR DESCRIPTION
**Pull request by naoyak**
Ref: #1090 

This aims to clean up some of the Python 2/3 compatibility code by using `six.*` and `six.moves.*` whenever possible. Some of the more complicated logic (e.g. using `StringIO`) has been omitted but may be added later.

### CODE REVIEW

**Comment by stevenbird** [comment]
cf https://github.com/nltk/nltk/pull/1485

**Comment by stevenbird** [comment]
@fievelk – since you've recently been working with six in NLTK, would you please review this?

**Comment by naoyak** [comment]
I think I'll stop for now before it gets too crazy. Tests are passing.

**Comment by naoyak** [comment]
Anyone able to review this PR? It deletes a fair chunk of code from `compat.py` and as such I think it's pretty useful.

**Comment by fievelk** [comment]
@naoyak I would ask you to please rebase the `develop` branch in your PR branch, so that we can check whether tests are passing. Current failures are related to `gensim`, and I think they might depend on previous errors that are now fixed.

**Comment by fievelk** [comment]
@kmike @dimazest Could you please help me reviewing this PR? I don't want to miss anything important here, since it is a pretty big PR.

**Comment by naoyak** [comment]
@fievelk thanks for pointing out the `gensim` test - actually, it looks like it's caused by a new major version of [gensim being released today on PyPI (1.0.0)](https://pypi.python.org/pypi/gensim). So I think this code needs to be refactored separately in any case.

```
Failed doctest test for gensim.doctest
  File "/scratch/jenkins/workspace/pull_request_tests/TOXENV/py34-jenkins/jdk/jdk8latestOnlineInstall/nltk/test/gensim.doctest", line 0

----------------------------------------------------------------------
File "/scratch/jenkins/workspace/pull_request_tests/TOXENV/py34-jenkins/jdk/jdk8latestOnlineInstall/nltk/test/gensim.doctest", line 48, in gensim.doctest
Failed example:
    model = gensim.models.Word2Vec.load_word2vec_format(word2vec_sample, binary=False)
Exception raised:
    Traceback (most recent call last):
      File "/scratch/jenkins/python/python-3.4.2-x86_64/lib/python3.4/doctest.py", line 1324, in __run
        compileflags, 1), test.globs)
      File "<doctest gensim.doctest[9]>", line 1, in <module>
        model = gensim.models.Word2Vec.load_word2vec_format(word2vec_sample, binary=False)
      File "/scratch/jenkins/workspace/pull_request_tests/TOXENV/py34-jenkins/jdk/jdk8latestOnlineInstall/.tox/py34-jenkins/lib/python3.4/site-packages/gensim/models/word2vec.py", line 1307, in load_word2vec_format
        raise DeprecationWarning("Deprecated. Use gensim.models.KeyedVectors.load_word2vec_format instead.")
    DeprecationWarning: Deprecated. Use gensim.models.KeyedVectors.load_word2vec_format instead.
----------------------------------------------------------------------
File "/scratch/jenkins/workspace/pull_request_tests/TOXENV/py34-jenkins/jdk/jdk8latestOnlineInstall/nltk/test/gensim.doctest", line 52, in gensim.doctest
Failed example:
    len(model.vocab)
Exception raised:
    Traceback (most recent call last):
      File "/scratch/jenkins/python/python-3.4.2-x86_64/lib/python3.4/doctest.py", line 1324, in __run
        compileflags, 1), test.globs)
      File "<doctest gensim.doctest[10]>", line 1, in <module>
        len(model.vocab)
    AttributeError: 'Word2Vec' object has no attribute 'vocab'
```

**Comment by fievelk** [comment]
@naoyak Thank you for pointing that out! I just submitted a new PR (#1638) to fix the problem.

**Comment by naoyak** [comment]
awesome, once that is merged I can rebase again.

**Comment by naoyak** [comment]
@fievelk thanks for the Gensim PR. This is rebased, let's see if tests pass.

**Comment by naoyak** [comment]
Tests passing now.

**Comment by fievelk** [comment]
[CI: retest]

File: nltk/app/srparser_app.py [hunk 1]
```diff
@@ -59,9 +59,7 @@
       [h]\t Help
       [Ctrl-p]\t Print
       [q]\t Quit
-"""
 
-"""
```
Comment (kmike) [inline nltk/app/srparser_app.py]: I'm not sure it was intended to have "Possible future improvements" as a module docstring; it looks more like a comment.

**Comment by kmike** [comment]
I haven't checked tkinter changes; all other changes look good! 👍

File: nltk/app/srparser_app.py (hunk 1 above)
Comment (naoyak) [reply nltk/app/srparser_app.py]: hmm yeah good point. These comments are over a decade old and I'm not sure they still apply... @stevenbird?

Comment (stevenbird) [reply nltk/app/srparser_app.py]: @naoyak: would you mind moving these to an inline comment? Thanks.

Comment (naoyak) [reply nltk/app/srparser_app.py]: @stevenbird: moved the comment to its own block below the imports.

**Comment by naoyak** [comment]
Comment addressed

**Comment by naoyak** [comment]
@stevenbird @kmike @fievelk is there anything else that needs to be reviewed here? Should the tkinter parts be factored out?

**Comment by stevenbird** [comment]
@naoyak – thanks for all your work here.
@kmike, @fievelk – thanks for the review.

(I'm less concerned about tkinter and we may end up removing some or all of it if it is too much work to maintain.)

**Pull request merged**